### PR TITLE
RepoInfo structure now includes new field representing RepoSummary

### PR DESCRIPTION
**Pull request by alexstan12**
**What type of PR is this?**
feature

**Which issue does this PR fix**:
ExpandedRepoInfo currently returns RepoInfo that is a list of Manifests.
To comply with the newest UI requirements, a new field called Summary,
referring to RepoSummary structure, was added.

**What does this PR do / Why do we need it**:
UI

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

This PR closes [#644](https://github.com/project-zot/zot/issues/664)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andaaron — APPROVED** [review]

File: pkg/extensions/search/schema.resolvers.go [hunk 1]
```diff
@@ -333,6 +333,31 @@ func (r *queryResolver) ExpandedRepoInfo(ctx context.Context, repo string) (*gql
...
+		platform := platform
+
+		summary.Platforms = append(summary.Platforms, &gql_generated.OsArch{
+			Os:   &platform.Os,
+			Arch: &platform.Arch,
+		})
+	}
+
+	summary.Size = &origRepoInfo.Summary.Size
+	// summary.Vendors = &origRepoInfo.Summary.Vendors
```
[the remark this answers is not in the document]
Comment (alexstan12) [reply pkg/extensions/search/schema.resolvers.go:352]: hmm, just a leftover. I'll remove it.

**Review by andaaron — APPROVED** [review]
Shall we rename some of the fields from https://github.com/project-zot/zot/pull/696/files#diff-84af1556304ab2f0edbb3541ee90f5e8eb88e4dde65c0a8069e517dea758c210R125 in this PR of separately (thinking of `Manifests` under RepoInfo)

**Comment by alexstan12** [comment]
> Shall we rename some of the fields from https://github.com/project-zot/zot/pull/696/files#diff-84af1556304ab2f0edbb3541ee90f5e8eb88e4dde65c0a8069e517dea758c210R125 in this PR or separately (thinking of `Manifests` under RepoInfo)
Since this replacement of 'manifests' with 'imageSummary' would inflict lots of changes in the ExpandedRepoInfo logic, I suggest fixing it in another PR.

**Comment by andaaron** [comment]
Gave the wrong link: https://github.com/project-zot/zot/pull/696/files#diff-84af1556304ab2f0edbb3541ee90f5e8eb88e4dde65c0a8069e517dea758c210R23
but ok, let's do it in a separate PR.

**Pull request merged**